### ISSUE TITLE
Two optimizers for linear and DNN separately

### ISSUE THREAD

**Issue by mangohero1985** [jrzaurin/Wide-and-Deep-Keras#2]
Hi Jrzaurin
Thank you so much for your excellent coding for w&d model.
I know this one you have finished two years ago, but I still want to talk about some details with you.

From the original paper and the official TF  implementation, the optimization was done by two optimizers parallel.  
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py#L328

Do you know how Keras can use two optimizers before concatenated the logits? Or Do you consider two optimizers are unnecessary?

**Comment by jrzaurin** [comment]
Hey there, sorry it took me this time to reply! 

I am at the moment implementing that solution in the Pytorch repo (i.e. ability to include two optimisers). Regarding to Keras, I am not a Keras expert...I guess one would have to code a customised optimiser to be applied to the unscaled outputs coming from the deep and the wide side separately, and then compile. How to precisely do that, I am not sure, if I find the solution I will include it in the code, and use it as an excuse to clean it up. 

Also 1) I am not sure there is a need for two optimisers (I did not manage to get that reasoning from the paper) and 2) I don't think FTRL is implemented in Keras (but correct me if I am wrong). Bear in mind that my idea was to implement a simple solution while recommending using the original one 😉  .

thanks for your suggestion.

**Comment by mangohero1985** [comment]
Hi Jrzaurin,
Thank you for the kindly reply.
I tried to implement a multiple optimizers feeding by myself in Keras.
The trainable weights can be arranged by "name" to different optimizers . 
Later, I want to make a pull request to Keras community.
You are right, there isn't FTRL implementation now in Keras. 
But, it is a good beginning, I will try to find a way to use it in Keras.
Thank you again.

**Comment by jrzaurin** [comment]
Cool, you can open a PR here and I merge it ;)

**Comment by andrewferlitsch** [comment]
If you reimplement using tf.keras, the TF team implemented the FTRL optimizer:
https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Ftrl

**Comment by jrzaurin** [comment]
Hey @andrewferlitsch Thanks for the suggestion. It actually would not be a bad idea to put use tf.keras and even move it to tf2.0. Maybe there is better to implement the 2 optimisers.

**Comment by atbolsh** [comment]
Basically, you have to overwrite the trainig_step and maybe the compile function. I found this guide illuminating:

https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit/
